pkg/issuer/venafi/setup.go

```diff
@@ -21,37 +21,39 @@ import (
 	"fmt"
 
 	apiutil "github.com/jetstack/cert-manager/pkg/api/util"
-	"github.com/jetstack/cert-manager/pkg/apis/certmanager/v1"
+	cmapi "github.com/jetstack/cert-manager/pkg/apis/certmanager/v1"
 	cmmeta "github.com/jetstack/cert-manager/pkg/apis/meta/v1"
 	logf "github.com/jetstack/cert-manager/pkg/logs"
 	corev1 "k8s.io/api/core/v1"
 )
 
-func (v *Venafi) Setup(ctx context.Context) error {
+func (v *Venafi) Setup(ctx context.Context) (err error) {
+	defer func() {
+		if err == nil {
```

Comment:
Might be my taste but it feels weird for the happy path to be in the if statement, i would personally reverse it also putting the smaller code block inside the `if` statement

Reply:
Yeah, I don't know why I wrote it like that. Now simplified.